core/client/fs/src/test/java/alluxio/client/file/BaseFileSystemTest.java

```diff
@@ -90,11 +90,11 @@ public DummyAlluxioFileSystem(FileSystemContext fsContext) {
   public void before() {
     mClientContext = ClientContext.create(mConf);
     mFileContext = PowerMockito.mock(FileSystemContext.class);
-    mFileSystem = new DummyAlluxioFileSystem(mFileContext);
     mFileSystemMasterClient = PowerMockito.mock(FileSystemMasterClient.class);
     when(mFileContext.acquireMasterClient()).thenReturn(mFileSystemMasterClient);
     when(mFileContext.getClientContext()).thenReturn(mClientContext);
     when(mFileContext.getConf()).thenReturn(mConf);
+    mFileSystem = new DummyAlluxioFileSystem(mFileContext);
```

Comment:
why does this order matter?

Reply:
because the BaseFileSystem now has an AlluxioBlockStore member which needed to return the proper client, conf, and context when creating the object